src/main/java/it/gov/pagopa/pu/bff/connector/process_executions/client/IngestionFlowFileSearchClient.java

```diff
@@ -0,0 +1,64 @@
+package it.gov.pagopa.pu.bff.connector.process_executions.client;
+
+import it.gov.pagopa.pu.bff.connector.process_executions.config.ProcessExecutionsApisHolder;
+import it.gov.pagopa.pu.bff.dto.IngestionFlowFileFiltersDTO;
+import it.gov.pagopa.pu.processexecutions.dto.generated.PagedModelIngestionFlowFile;
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+@Service
+@Slf4j
+public class IngestionFlowFileSearchClient {
+
+    private final ProcessExecutionsApisHolder processExecutionsApisHolder;
+
+    public IngestionFlowFileSearchClient(
+      ProcessExecutionsApisHolder processExecutionsApisHolder) {
+        this.processExecutionsApisHolder = processExecutionsApisHolder;
+    }
+
+  public PagedModelIngestionFlowFile getIngestionFlowFiles(IngestionFlowFileFiltersDTO ingestionFlowFileFilters, String operatorExternalId, Pageable pageable, String accessToken) {
+    try {
+      Integer pageNumber = pageable.isPaged()? pageable.getPageNumber() : 0;
+      Integer pageSize = pageable.isPaged()? pageable.getPageSize() : null;
+      String organizationId = ingestionFlowFileFilters != null ? String.valueOf(
+        ingestionFlowFileFilters.getOrganizationId()) : null;
+      String flowFileType = ingestionFlowFileFilters != null
+        ? ingestionFlowFileFilters.getFlowFileType() : null;
+      OffsetDateTime creationDateFrom = ingestionFlowFileFilters != null
+        ? ingestionFlowFileFilters.getCreationDateFrom() : null;
+      OffsetDateTime creationDateTo = ingestionFlowFileFilters != null
+        ? ingestionFlowFileFilters.getCreationDateTo() : null;
+      String status =
+        ingestionFlowFileFilters != null ? ingestionFlowFileFilters.getStatus()
+          : null;
+      String fileMame = ingestionFlowFileFilters != null
+        ? ingestionFlowFileFilters.getFileName() : null;
```

Comment:
why are you checking if inglestionFlowFileFilters is null?

Reply:
I see only after that you have introduced a DTO for filters!
It should not be null! some filters are mandatory ;)